activestorage/lib/active_storage/attached/changes/create_one.rb

```diff
@@ -70,6 +70,28 @@ def find_or_build_blob
           )
         when String
           ActiveStorage::Blob.find_signed!(attachable, record: record)
+        when File
```

Comment:
The `File` and `Pathname` branches have almost identical structures. Would it be worth trying to combine them?

```ruby
when File, Pathname
  io = attachable.is_a?(File) ? attachable : attachable.open
  filename = File.basename(attachable.path)
  content_type = Marcel::MimeType.for(io, name: filename)

  ActiveStorage::Blob.create_and_upload!(
    io: io,
    filename: filename,
    content_type: content_type,
    record: record,
    service_name: attachment_service_name
  )
end
```

Reply:
I think this would be a good refactoring!

Reply:
I've applied the refactoring and ran the tests and they pass